pkg/httpmetrics/disk_metrics.go

```diff
@@ -0,0 +1,90 @@
+package httpmetrics
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/chainguard-dev/clog"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/shirou/gopsutil/disk"
+)
+
+const (
+	DiskUsageScrapeInterval = 5 * time.Second
+)
+
+var (
+	// Prometheus metrics for disk usage.
+	diskUsageBytesGauge = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "disk_usage_bytes",
+			Help: "Disk usage in bytes.",
+		},
+		[]string{"mount"},
+	)
+
+	// So that we only start the disk usage scraper once.
+	once = new(sync.Once)
+)
+
+// scrapeDiskUsage returns the disk usage of all mounted partitions.
+//
+// On Cloud Run, there are 4 partitions:
+// - /tmp
+// - /
+// - /dev
+// - /dev/shm
+// - and any additional volume mounts.
+//
+// We ignore all the /dev partitions here, keeping /, /tmp, and any additional
+// volumn mounts that may come from the Revision config.
+func scrapeDiskUsage() map[string]uint64 {
+	parts, err := disk.Partitions(true)
+	if err != nil {
+		// It is better to be silent here and missing metrics, than to be spam log
+		// here, and/or panic.
+		return nil
```

Comment:
im fine not logging, 
can we add a metric for failures so we know how often its failing?

Reply:
Done